### PR TITLE
Add tutorial for RL GRPO gemma4-e4b on v6e 4x8

### PR DESCRIPTION
**Pull request by dasoto**
# Description

Adding tutorial for RL GRPO gemma-4-e4b on v6e 4x8 cluster using xpk.

# Tests

Please describe how you tested this change, and include any instructions and/or
commands to reproduce.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/examples/chat_templates/openmathinstruct2_rl.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "SYSTEM_PROMPT": "You are given a problem. Think about the problem and provide your reasoning. Place it between {reasoning_start_token} and {reasoning_end_token}. Then, provide the final answer (i.e., just one numerical value) between {solution_start_token} and {solution_end_token}.",
```
Comment (A9isha) [inline src/maxtext/examples/chat_templates/openmathinstruct2_rl.json:2]: could you please name this file following the pattern of other files in this directory?

File: scripts/run_gemma4_e4b_hf_to_maxtext.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+CONVERTED_CKPT_BASE_DIR="${BASE_OUTPUT_DIRECTORY}/checkpoints/gemma4-e4b-converted"
+echo "Converted checkpoint will be saved to: ${CONVERTED_CKPT_BASE_DIR}"
+
+# Run the conversion script.
+# NOTE: scan_layers must be False for the gemma4_small decoder block (E2B/E4B);
+# scanned-layer conversion is not supported for this block.
+# NOTE: --hf_model_path pins the conversion to the BASE model (google/gemma-4-E4B).
+# Without it, MaxText defaults to HF_IDS[gemma4-e4b] = google/gemma-4-E4B-it (the
+# instruction-tuned model), which is not what we want for this base-model RL run.
+python3 -m maxtext.checkpoint_conversion.to_maxtext \
```
Comment (A9isha) [inline scripts/run_gemma4_e4b_hf_to_maxtext.sh]: just for keeping parity with other files, we might not need the separate file

File: scripts/run_gemma4_e4b_maxtext_to_hf.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    echo "Error: BASE_OUTPUT_DIRECTORY is not set. Please set it in the script or as an environment variable."
+    exit 1
+fi
+
+if [ -z "$MAXTEXT_CKPT_PATH" ]; then
+    echo "Error: MAXTEXT_CKPT_PATH is not set. Please set it in the script or as an environment variable."
+    exit 1
+fi
+
+# Install torch for conversion
```
Comment (A9isha) [inline scripts/run_gemma4_e4b_maxtext_to_hf.sh]: just for keeping parity with other files, we might not need the separate file

Comment (dasoto) [reply scripts/run_gemma4_e4b_maxtext_to_hf.sh]: Not sure that I understand this comment. qwen3_30b have also two files a hf_to_maxtext and maxtext_to_hf

File: src/maxtext/examples/chat_templates/openmathinstruct2_rl.json (hunk 1 above)
Comment (dasoto) [reply src/maxtext/examples/chat_templates/openmathinstruct2_rl.json:2]: Done

File: scripts/run_gemma4_e4b_maxtext_to_hf.sh (hunk 3 above)
Comment (A9isha) [reply scripts/run_gemma4_e4b_maxtext_to_hf.sh]: sorry about the confusion, please take a look at this [PR](https://github.com/AI-Hypercomputer/maxtext/pull/4463)

Comment (dasoto) [reply scripts/run_gemma4_e4b_maxtext_to_hf.sh]: Thanks Anisha, I modified everyrhtin to follow the new structure and format.

**Review by A9isha — APPROVED** [review]
Thanks David!

**Review by SurbhiJainUSC — APPROVED** [review]

File: src/maxtext/trainers/post_train/rl/scripts/run_gemma4_e4b_rl.sh [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+batch_size=128 \
+num_batches=500 \
+learning_rate_schedule_steps=500 \
+num_test_batches=5 \
+eval_interval=100 \
+rl.num_generations=8 \
+rl.num_iterations=1 \
+rl.grpo_beta=0.05 \
+rl.grpo_epsilon=0.2 \
+debug.rl=True \
```
Comment (SurbhiJainUSC) [inline src/maxtext/trainers/post_train/rl/scripts/run_gemma4_e4b_rl.sh:130]: FYI: This config has been changed to `debug` in main: https://github.com/AI-Hypercomputer/maxtext/pull/4431

Comment (dasoto) [reply src/maxtext/trainers/post_train/rl/scripts/run_gemma4_e4b_rl.sh:130]: Will fix in next PR whenadd the tutorial for 26b
